core/CronArchive/Performance/Logger.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\CronArchive\Performance;
+
+use Piwik\ArchiveProcessor;
+use Piwik\Common;
+use Piwik\Config;
+use Piwik\Option;
+use Piwik\Timer;
+use Piwik\Url;
+use Psr\Log\LoggerInterface;
+
+// TODO: need scheduled task to clean up old ones? maybe cronarchive should delete them
```

Comment:
is it needed to store the messages in the option table at all? I thought maybe it's only useful to have the profiling info in the log when configured to log at this level.

Reply:
In order to display it in the core:archive output, it needs to be stored somewhere. Could put it in a tmp file also, doesn't have to be an option.

Reply:
Random idea... not sure if it works... maybe we could log such information into a log file? So instead of enabling/disabling the feature... we would configure a path to the log file? This can be especially useful when writing the cron output into a log file anyway like `./console core:archive > archive.log`.

Reply:
@tsteur so `./console core:archive --log-performance=./archive-performance.log`?

Reply:
I was more thinking `archive_profiling_log="/home/path/foo.log"` (with an absolute path if possible) so we don't need to pass through the path to the log file to the actual archiving request through http etc which could introduce a security issue maybe (eg imagine writing into /etc/passwd or so just as an example). Although `--log-performance` has for sure some advantages in that it can be easily enabled/disabled and it allows usage of parameters like date in the log file etc.